packages/typespec-vscode/src/vscode-cmd/emit-code/emit-code.ts

```diff
@@ -287,60 +312,75 @@ async function doEmit(mainTspFile: string, emitter: Emitter) {
   /*Config emitter output dir and emit in tspconfig.yaml. */
   const defaultEmitOutputDirInConfig = `{output-dir}/{emitter-name}`;
   const tspConfigFile = path.join(baseDir, TspConfigFileName);
-  let configYaml = parseDocument(""); //generate a empty yaml
+  let configYaml = new Document(); //generate a empty yaml
   if (await isFile(tspConfigFile)) {
-    const content = await readFile(tspConfigFile);
-    configYaml = parseDocument(content.toString());
+    const content = await tryReadFile(tspConfigFile);
+    if (content) {
+      configYaml = tryParseYaml(content.toString()) ?? configYaml;
+    }
   }
-  let outputDir = defaultEmitOutputDirInConfig;
+
+  const generations: {
+    emitter: Emitter;
+    outputDir: string;
+    options?: Record<string, string>;
+    codeInfo: string;
+  }[] = [];
   try {
-    /*update emitter in config.yaml. */
-    const emitNode = configYaml.get("emit");
-    if (emitNode) {
-      if (isSeq(emitNode)) {
-        if (Array.isArray(emitNode.items)) {
-          const index = emitNode.items.findIndex((item) => {
-            if (isScalar(item)) {
-              return item.value === emitter.package;
+    for (const emitter of emitters) {
+      let outputDir = defaultEmitOutputDirInConfig;
+      /*update emitter in config.yaml. */
+      const emitNode = configYaml.get("emit");
+      if (emitNode) {
+        if (isSeq(emitNode)) {
+          if (Array.isArray(emitNode.items)) {
+            const index = emitNode.items.findIndex((item) => {
+              if (isScalar(item)) {
+                return item.value === emitter.package;
+              }
+              return false;
+            });
+            if (index === -1) {
+              emitNode.items.push(emitter.package);
             }
-            return false;
-          });
-          if (index === -1) {
-            emitNode.items.push(emitter.package);
           }
         }
+      } else {
+        configYaml.set("emit", [emitter.package]);
       }
-    } else {
-      configYaml.set("emit", [emitter.package]);
-    }
-    const emitOutputDir = configYaml.getIn(["options", emitter.package, "emitter-output-dir"]);
-    if (!emitOutputDir) {
-      configYaml.setIn(
-        ["options", emitter.package, "emitter-output-dir"],
-        defaultEmitOutputDirInConfig,
-      );
-    } else {
-      outputDir = emitOutputDir as string;
+      const emitOutputDir = configYaml.getIn(["options", emitter.package, "emitter-output-dir"]);
```

Comment:
Can we not call `resolveCompilerOptions` to not reinvent interpolation here that will most likely get out of date.

Reply:
this also does look liek it wouldn't work if you are interpolating user parameters/environment variables

Reply:
> Can we not call `resolveCompilerOptions` to not reinvent interpolation here that will most likely get out of date.

Hello @timotheeguerin Do  you suggest to call `resolveCompilerOptions` here to get out the emitter options?
Here we just parse the tspconfig.yaml file to get pre-config emitter and options, and will update the tspconfig.yaml after config the emitter. We will not retrieve the options via `resolveCompilerOptions` which may get the default value if one emitter option is not defined in the tspconfig.yaml (e.g. emitter-output-dir: ./tsp-output/@typespec/http-client-csharp, and we will not set this value back to tspconfig.yaml

Reply:
You just want to figure out the output for here right? You can ignore the rest?
With your logic as said above it will be broken if the user defined parameters in the config and use them in the output dir value.

Reply:
hello @timotheeguerin , Here we don't figure out the path of output directory, we just check if there is `emitter-output-dir` option is defined in tspconfig.yaml, the value maybe a real path or a path with parameter. We just make sure that it is defined in tspconfig.yaml, if not, we will set the option with default parameterized path ("{output-dir}/{emitter-name}") in the tspconfig.yaml. The purpose to add this emitter option in tspconfig.yaml is to tell customer there is this option to set the output dir they want.

Yes. we may need to resolve out real path (not parameterized one in tspconfig.yaml) and used it in vscode extension log. I will file another pr to enhance this.

Reply:
but doesn't the code below interpolate the value of the user to display it, that value is then potentially incorrect 

```ts
      
      outputDir = outputDir
        .replace("{project-root}", baseDir)
        .replace("{output-dir}", `${baseDir}/tsp-output`)
        .replace("{emitter-name}", emitter.package);
```

Reply:
Yes. this is used in the log which I will enhance in next pr to call `resolveCompilerOptions` to get the real path.